merkletree.go

```diff
@@ -0,0 +1,274 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+var (
+	ErrInvalidTree = errors.New("[merkletree] invalid tree")
+)
+
+const (
+	EmptyBranchIdentifier = 'E'
+	LeafIdentifier        = 'L'
+)
+
+type MerkleTree struct {
+	treeNonce []byte
+	salt      []byte
+	privKey   []byte
+	pubKey    []byte
+	policies  Policies
+	root      *interiorNode
+}
+
+func InitMerkleTree(policies Policies, treeNonce, salt, pubKey, privKey []byte) *MerkleTree {
+	leftBranch := &emptyNode{
+		node: node{
+			level: 1,
+		},
+	}
+
+	rightBranch := &emptyNode{
+		node: node{
+			level: 1,
+		},
+	}
+	root := &interiorNode{
+		node: node{
+			parent: nil,
+			level:  0,
+		},
+		leftChild:  leftBranch,
+		rightChild: rightBranch,
+		leftHash:   nil,
+		rightHash:  nil,
+	}
+	leftBranch.parent = root
+	rightBranch.parent = root
```

Comment:
This whole pattern of creating an `interiorNode` is repeated in `insertNode`.  It could benefit from a `NewInteriorNode` constructor that takes a `parent` and `level`.